### PR TITLE
Adds CI/CD overview page

### PR DESCRIPTION
**Pull request by darshan-nagaraj**
JIRA: https://issues.redhat.com/browse/OSDOCS-2562

OCP version: 4.6+

Direct Doc Preview Link: https://deploy-preview-44137--osdocs.netlify.app/openshift-enterprise/latest/cicd/index.html

@rolfedh @sounix000 @DebarghoGhosh PTAL at the PR and let me know your feedback.

@jitendar-singh Please review the contact and provide QA ack.

### CODE REVIEW

File: cicd/index.adoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
+:_content-type: ASSEMBLY
+[id="ci-cd-overview"]
+= {product-title} CI/CD overview
+include::_attributes/common-attributes.adoc[]
+:context: ci-cd-overview
+
+toc::[]
+
+
+{product-title} is an enterprise ready Kubernetes platform for developers, which enables organizations to automate application delivery process through DevOps practices, such as continuous integration (CI) and continuous delivery (CD). To meet your organizational needs, the {product-title} provides the following CI/CD solutions:
```
Comment (rolfedh) [inline cicd/index.adoc:10]:
```suggestion
{product-title} is an enterprise-ready Kubernetes platform for developers, which enables organizations to automate the application delivery process through DevOps practices, such as continuous integration (CI) and continuous delivery (CD). To meet your organizational needs, the {product-title} provides the following CI/CD solutions:
```

File: cicd/index.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+[id="openshift-pipelines"]
+== OpenShift Pipelines
+OpenShift Pipelines provides a Kubernetes-native CI/CD framework to design and run each step of the CI/CD pipeline in its own container. It can scale independently to meet the on-demand pipelines with predictable outcomes.
+
+For more information, see xref:../cicd/pipelines/understanding-openshift-pipelines.adoc#understanding-openshift-pipelines[Understanding OpenShift Pipelines]
+
+[id="openshift-gitops"]
+== OpenShift GitOps
+OpenShift GitOps is a Red Hat OpenShift operator that uses Argo CD as the declarative GitOps engine that enables GitOps workflows across multicluster OpenShift and Kubernetes infrastructure. GitOps defines infrastructure and application definitions as code. Then, it uses this code to manage multiple workspaces and clusters to simplify the creation of infrastructure and application configurations. GitOps also handles and automates complex deployments at a fast pace, saving time during deployment and release cycles.
```
Comment (rolfedh) [inline cicd/index.adoc:36]:
```suggestion
OpenShift GitOps is an Operator that uses Argo CD as the declarative GitOps engine. It enables GitOps workflows across multicluster OpenShift and Kubernetes infrastructure. GitOps defines infrastructure and application definitions as code. Then, it uses this code to manage multiple workspaces and clusters to simplify the creation of infrastructure and application configurations. GitOps also handles and automates complex deployments at a fast pace, saving time during deployment and release cycles.
```
I recommend further pruning beyond what I've suggested here. 

This paragraph feels problematic, overloaded, and mixes conceptual information with sequential/procedural information. 

The reader only needs a *short* overview here. They will go to the respective sections for more detailed information.

File: cicd/index.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+For more information, see xref:../cicd/gitops/understanding-openshift-gitops.adoc#understanding-openshift-gitops[Understanding OpenShift GitOps]
+
+[id="jenkins-ci-cd"]
+== Jenkins
+Jenkins automates the process of building, testing, and deploying applications and projects. OpenShift Developer Tools provides a Jenkins image that integrates directly with the {product-title}. Jenkins can be deployed on OpenShift using the templates provided by the Samples Operator, or with the certified Helm chart.
```
Comment (rolfedh) [inline cicd/index.adoc:42]:
```suggestion
Jenkins automates the process of building, testing, and deploying applications and projects. OpenShift Developer Tools provides a Jenkins image that integrates directly with the {product-title}. Jenkins can be deployed on OpenShift by using the Samples Operator templates or certified Helm chart.
```

File: cicd/index.adoc (hunk 1 above)
Comment (darshan-nagaraj) [reply cicd/index.adoc]: @rolfedh Fixed.

File: cicd/index.adoc (hunk 3 above)
Comment (darshan-nagaraj) [reply cicd/index.adoc]: @rolfedh Done.

File: cicd/index.adoc (hunk 2 above)
Comment (darshan-nagaraj) [reply cicd/index.adoc]: @rolfedh Please take a look and let me know if it makes sense ?

> OpenShift GitOps is an Operator that uses Argo CD as the declarative GitOps engine. It enables GitOps workflows across multicluster OpenShift and Kubernetes infrastructure. Using OpenShift GitOps, administrators can consistently configure and deploy Kubernetes-based infrastructure and applications across clusters and development lifecycles.

**Review by rolfedh — APPROVED** [review]

**Comment by bergerhoffer** [comment]
@mburke5678 these CPs are still open, can you merge them?

@darshan-nagaraj Looks like the CP to 4.6 failed, can you please do a manual CP for it?

**Pull request merged**